### DISCUSSION TITLE
Questions regarding premining account balances and the native coin of a Polygon SDK blockchain

### DISCUSSION THREAD

**Discussion by AzulPretoBranco** [0xPolygon/polygon-edge#298] [Q&A] [answered]
While reading the documentation (https://sdk-docs.polygon.technology/docs/get-started/set-up-ibft-locally) I came across this statement:

![image](https://user-images.githubusercontent.com/96196879/146763905-4786dbee-1328-4286-bec4-af83219fa092.png)

I have a couple of questions regarding this.

1 - Does this mean that the native coin of a Polygon SDK blockchain is ETH? Or is that just an example and we can basically name the coin whatever we want?

2 - Since this coin will be used to, for example, pay for gas, how can we mine this coin after the blockchain is running? My understanding is that Polygon SDK blockchain uses PoA and not PoW, in that case is there a way for us to mine this coin? If not what will happen when the premined amount is exhausted? Will we stop being able to execute transactions because we have no way to pay for gas?

Since the documentation is not clear about this can someone please help clarify these questions?

**Comment by mrwillis** [comment, marked as answer]
Chiming in here to help:

1. You can name the token whatever you want. Actually this is a client level thing. For example in metamask when you add a network you can specify the "symbol" of the native currency. SDK is agnostic to this

2. There's no mining or rewards in the SDK - you will have to do that off-protocol (maybe have a separate account that will send rewards for stakers/proposers for proposing blocks, for example). In general I don't think there is mechanisms to increase supply @lazartravica @zivkovicmilos

**Reply by AzulPretoBranco** [reply]
Hi @mrwillis, thanks so much for the reply!

In that case, if there is no way to increase supply in the SDK, I assume that I would have to generate all the native tokens that will be needed in the future when starting the blockchain for the first time (when creating the genesis file).

Let's say I'm planning on building a chain that will last around 10 years (just an example). I would have to generate a sufficient amount of native tokens when creating the genesis file so I can make sure that I will be able to pay for gas (among other things) in the next 10 years.

Am I thinking correctly?

**Reply by mrwillis** [reply]
Yes that's right

**Reply by jooddang** [reply]
@mrwillis Hi, thanks for the answer. In my understanding, you, as a user who executes a contract like swap or transfer, have to pay gas fee with native token (that can only be earned at genesis). 
1. At first, how does a user earn native tokens for gas?
2. If native token can't be generated after genesis, once all native token is burned by gas fee, how does the network work?

**Reply by samisbakedham** [reply]
> @mrwillis Hi, thanks for the answer. In my understanding, you, as a user who executes a contract like swap or transfer, have to pay gas fee with native token (that can only be earned at genesis). 
> 
> 1. At first, how does a user earn native tokens for gas?
> 
> 2. If native token can't be generated after genesis, once all native token is burned by gas fee, how does the network work?

Native tokens that are "burned" as gas fees are just transferred to the validator nodes I think.

**Reply by jooddang** [reply]
does it mean that reserve pool or validators wallets gonna be the only source for native token, right? @samisbakedham
